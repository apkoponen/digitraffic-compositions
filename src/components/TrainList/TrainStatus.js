import React, { Component } from 'react';
import './TrainStatus.css';

class TrainStatus extends Component {
  render() {
    const statusClass = this.props.running ? 'running' : 'stopped';
    const statusText = this.props.running ? 'Moving' : 'Stopped';

    return (
      <div className={`TrainStatus ${statusClass}`}>
        <span className="TrainStatus__dot" />
        {statusText}
      </div>
    );
  }
}

export default TrainStatus;
