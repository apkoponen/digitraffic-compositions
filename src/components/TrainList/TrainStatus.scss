@import 'styles/_colors';

$color-running: #29a21b;
$color-stopped: #8B0000;

.TrainStatus {
  text-transform: uppercase;
  font-size: 12px;

  &.running {
    color: $color-running;
    .TrainStatus__dot {
      background: $color-running;
    }
  }
  &.stopped {
    color: $color-stopped;
    .TrainStatus__dot {
      background: $color-stopped;
    }
  }
}

.TrainStatus__dot {
  display: inline-block;
  width: 8px;
  height: 8px;
  vertical-align: baseline;
  border-radius: 50%;
  margin-right: 3px;
}