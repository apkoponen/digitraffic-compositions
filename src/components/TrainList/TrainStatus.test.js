import React from 'react';
import TrainStatus from './TrainStatus';
import renderer from 'react-test-renderer';

it('renders running status correctly', () => {
  const tree = renderer.create(<TrainStatus running />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders stopped status correctly', () => {
  const tree = renderer.create(<TrainStatus />).toJSON();
  expect(tree).toMatchSnapshot();
});
