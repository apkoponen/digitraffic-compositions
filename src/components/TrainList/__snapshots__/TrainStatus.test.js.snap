// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders running status correctly 1`] = `
<div
  className="TrainStatus running"
>
  <span
    className="TrainStatus__dot"
  />
  Moving
</div>
`;

exports[`renders stopped status correctly 1`] = `
<div
  className="TrainStatus stopped"
>
  <span
    className="TrainStatus__dot"
  />
  Stopped
</div>
`;
